client/allocrunner/taskrunner/task_runner.go

```diff
@@ -650,12 +668,17 @@ func (tr *TaskRunner) persistLocalState() error {
 // buildTaskConfig builds a drivers.TaskConfig with an unique ID for the task.
 // The ID is consistently built from the alloc ID, task name and restart attempt.
 func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
+
 	return &drivers.TaskConfig{
 		ID:   fmt.Sprintf("%s/%s/%d", tr.allocID, tr.taskName, tr.restartTracker.GetCount()),
 		Name: tr.task.Name,
 		Resources: &drivers.Resources{
 			NomadResources: tr.task.Resources,
-			//TODO Calculate the LinuxResources
+			LinuxResources: &drivers.LinuxResources{
+				MemoryLimitBytes: int64(tr.Task().Resources.MemoryMB) * 1024 * 1024,
+				CPUShares:        int64(tr.Task().Resources.CPU),
+				PercentTicks:     float64(tr.task.Resources.CPU) / float64(tr.clientConfig.Node.Resources.CPU),
```

Comment:
Peeking at the comments in [structs.go](https://github.com/hashicorp/nomad/blob/206d98cf0dc3289f5222e1f11b3c63560e493d9b/nomad/structs/structs.go#L1443-L1444) I think you want to use `NodeResources`, but I'm not positive.

We really need to update the comments in `structs.go` to instruct developers on which field to use. I'm assuming since `Resources` is marked deprecated we shouldn't use it?